### PR TITLE
fix(files): Fix newly created folder not appearing in main files view

### PR DESCRIPTION
**Pull request by Pytal**
A `files:node:created` event is emitted correctly but was missing the fileid

image

For #1285

### CODE REVIEW

File: lib/composables/dav.ts [hunk 1]
```diff
@@ -85,8 +85,26 @@ export const useDAVFiles = function(
 	async function createDirectory(name: string): Promise<Folder> {
 		const path = join(currentPath.value, name)
 
-		await client.value.createDirectory(join(defaultRootPath.value, path))
-		const directory = await getFile(path) as Folder
+		const { headers } = await client.value.customRequest(
```
Comment (susnux) [inline lib/composables/dav.ts:88]: Why a custom request? The previously used `createDirectory` should work, if not then something else is broken.

File: lib/composables/dav.ts [hunk 2]
```diff
@@ -85,8 +85,26 @@ export const useDAVFiles = function(
...
+			join(defaultRootPath.value, path),
+			{
+				method: 'MKCOL',
+			},
+		)
+		const stat = await getFile(path)
+		const directory = new Folder({
```
Comment (susnux) [inline lib/composables/dav.ts:95]: `getFile` already returns a Node so in this case a Folder. If this does not work then `@nextcloud/files` is broken

**Comment by emoral435** [comment]
Updated comment to show redirection between chained PR's :)

File: lib/composables/dav.ts (hunk 1 above)
Comment (Pytal) [reply lib/composables/dav.ts:88]: Used a custom request here to be able to access the headers as `createDirectory` doesn't return them

**Pull request closed without merging**